### DISCUSSION TITLE
Connection closed when using computeWithDatabase

### DISCUSSION THREAD

**Discussion by josh-burton** [simolus3/drift#2331] [Q&A]
I'm using WorkManager to execute some code on a background isolate, and attempting to use Drift's isolates feature to share a database connection via isolates.

In my app foreground isolate, I use Drift normally:

```
database = SyncLogDatabase(SyncLogDatabase.openLazyConnection());

database.select(myTable)....etc
```

In my background isolate, I use computeWithDatabase:

```
  _database = SyncLogDatabase(SyncLogDatabase.openLazyConnection());

   await _database.computeWithDatabase(
        computation: (database) async {
          await database.into(myTable)...;
        },
        connect: SyncLogDatabase.new,
      );
```

But whenever this background isolate attempts to modify some data, I get the following error:

`StatementMethod.deleteOrUpdate: over isolate channel, but the connection was closed!`

I feel like I must be doing something wrong. Is this the correct usage of computeWithDatabase?

**Comment by simolus3** [comment]
It's hard to say for sure what's going wrong here, my best guess so far is that there might be an unawaited async operation on the database. We close the isolate channel after the callback for `computeWithDatabase` returns, so if there's any pending operation remaining, I can see that exception happening.

Do you only have that single insert in `computeWithDatabase` or was that just an example? It's suspicious because the error references `StatementMethod.deleteOrUpdate`, which is not used for running inserts.

**Reply by simolus3** [reply]
FWIW a similar issue was fixed in `drift: 1.6.0`, so maybe this problem has been fixed as well?

**Reply by josh-burton** [reply]
Ok great. I will close this until I come across the issue again. Thanks for the help.
